MaxText/layers/linears.py

```diff
@@ -315,6 +326,69 @@ def generate_kernels(self, num_experts, base_emb_dim, mlp_dim):
     wo_kernel = jnp.asarray(wo_kernel, self.dtype)
     return w0_kernel, w1_kernel, wo_kernel
 
+  def permute(self, inputs, gate_logits, emb_dim):
```

Comment:
Can you include a comment here for what this function does? I'm not sure if permute is the best name - maybe like get_inputs_for_experts?

Reply:
Thanks Matt! I was following the [paper](https://screenshot.googleplex.com/8dwYGgsy8WE6NDL) to name the method. Let me add some comments, and hopefully it will have less confusing?